dtlstransport.go

```diff
@@ -214,21 +240,26 @@ func (t *DTLSTransport) Start(remoteParameters DTLSParameters) error {
 		ClientAuth:             dtls.RequireAnyClientCert,
 		LoggerFactory:          t.api.settingEngine.LoggerFactory,
 	}
+
+	t.onStateChange(DTLSTransportStateConnecting)
```

Comment:
Is it possible to transition from failed to connecting? I know in a PeerConnection that state transition is forbidden but I'm not sure about ORTC.

Reply:
Do you think the right thing to do here is `if state == failed` that bails out?

Reply:
I don't know. Does the spec say? Couldnt find any more information.

Reply:
That sounds right to me https://github.com/pion/webrtc/pull/730/files#diff-af8b99e79ad0bc29625704279b2d48e5R230

Will read spec and see if they say anything

Reply:
`Start` doesn't exist for WebRTC so nothing there.

I would say this is the right behavior, I bet internally they do have the check.

Reply:
Matched ORTC here `If start is called after a previous start call, or if state is closed, throw an InvalidStateError.`